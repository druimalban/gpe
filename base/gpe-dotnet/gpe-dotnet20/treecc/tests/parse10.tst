// test C# operation definition errors

%option lang = "C#"

%node expr %typedef

%operation void op1(expr e)

op1(expr)
{
}

%operation void op2::op2(expr e)

op2(expr)
{
}

%operation void Op3::op3(expr e)

op3(expr)
{
}
