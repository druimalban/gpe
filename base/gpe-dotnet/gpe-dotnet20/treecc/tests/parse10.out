line 3: %option lang C#
line 5: %node expr no_parent 4
parse10.tst:7: C# requires that a class name be specified
line 7: %operation void op1 0
line 7: %param expr e 1
line 9: %case expr op1
parse10.tst:13: C# requires different class and operation names
line 13: %operation void op2::op2 0
line 13: %param expr e 1
line 15: %case expr op2
line 19: %operation void Op3::op3 0
line 19: %param expr e 1
line 21: %case expr op3
